### PR TITLE
v1.9: `distinct` search parameter

### LINKED ISSUE #2853: v1.9: Distinct attributes
v1.9 adds a new search parameter, `distinct`. This parameter accepts an attribute. The supplied attribute to `distinct` must be present in the `filterableAttributes` index setting. If multiple search results share values across one or more of these attributes, Meilisearch will only return the best ranked document.

This behaviour is similar to the `distinctAttribute` index setting, but can be configured at search time. `distinctAttribute` remains unchanged and acts as a default distinct attribute value that can be overridden by `distinct`.

## Tasks
- [ ] Update the search API reference
- [ ] 

## References
- PRD: https://www.notion.so/meilisearch/Set-Distinct-Attribute-at-Search-ebaaa195ac2149a1a592c9243dde6f5b
- Engine issue: https://github.com/meilisearch/meilisearch/issues/4611
- SME: @Kerollmops

### PR DESCRIPTION
**Pull request by guimachiavelli**
Closes #2853

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -1287,6 +1287,31 @@ analytics_event_bind_event_1: |-
...
+      "sku",
+      "url"
+    ]'
+distinct_attribute_guide_distinct_parameter_1: |-
+  curl \
+    -X POST 'http://localhost:7700/indexes/products/search' \
+    -H 'Content-Type: application/json' \
+    --data-binary '{
+      "q": "white shirt",
+      "distinct": "product_id"
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:1313]: I think SKU is a more common example for people demanding distinct, but maybe I'm wrong, so you can check with product team

Comment (guimachiavelli) [reply .code-samples.meilisearch.yaml:1313]: Sure, I can change it. It's a purely illustrative example showing how one would use it in a "real life" setting, so it's more of a question of it being somewhat credible on a quick glance.

Comment (curquiza) [reply .code-samples.meilisearch.yaml]: This is where product team can help you with a real use case 👍

File: learn/index_settings/distinct.mdx [hunk 2]
```diff
@@ -7,6 +7,12 @@ description: Distinct attribute is a field that prevents Meilisearch from return
 
 The distinct attribute is a special, user-designated field. It is most commonly used to prevent Meilisearch from returning a set of several similar documents, instead forcing it to return only one.
 
+You may set a distinct attribute in two ways: using the `distinctAttribute` index setting during configuration, or the `distinct` search parameter at search time.
+
+## Setting a distinct attribute during configuration
+
+`distinctAttribute` is an index setting that configures a default distinct attribute Meilisearch applies to all searches in that index.
```
Comment (Kerollmops) [inline learn/index_settings/distinct.mdx:14]:
```suggestion
`distinctAttribute` is an index setting that configures a default distinct attribute Meilisearch applies to all searches and facets retrieval in that index.
```

File: reference/api/search.mdx [hunk 3]
```diff
@@ -565,6 +565,20 @@ The response shows the facet distribution for `genres` and `rating`. Since `rati
 
 [Learn more about facet distribution in the faceted search guide.](/learn/fine_tuning_results/faceted_search#configuring-and-using-facets)
 
+### Distinct attributes at search time
+
+**Parameter**: `distinct`<br />
+**Expected value**: An `attribute` present in the `filterableAttributes` list<br />
+**Default value**: `null`
+
+Defines one attribute in the `filterableAttributes` list as a distinct attribute. Distinct attributes indicate documents sharing the same value for the specified field are equivalent and only one of them should be returned in search results.
```
Comment (Kerollmops) [inline reference/api/search.mdx:574]:
```suggestion
Defines one attribute in the `filterableAttributes` list as a distinct attribute. Distinct attributes indicate documents sharing the same value for the specified field are equivalent and only the most relevant one should be returned in search results.
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
LGTM but some precisions, thank you gui 😃

**Review by curquiza — APPROVED** [review]
Ok for code samples

**Pull request merged**